detectable/src/main/java/com/synopsys/integration/detectable/detectables/projectinspector/ProjectInspectorParser.java

```diff
@@ -33,18 +33,41 @@ public ProjectInspectorParser(Gson gson, ExternalIdFactory externalIdFactory) {
         this.externalIdFactory = externalIdFactory;
     }
 
-    public List<CodeLocation> parse(String inspectionOutput) {
-        ProjectInspectorOutput projectInspectorOutput = gson.fromJson(inspectionOutput, ProjectInspectorOutput.class);
+    public List<CodeLocation> parse(File outputFile) {
 
-        // If modules is not present in the output then project inspector has not found any open source dependencies.
-        // Return an empty list so callers do not fail when examining the results.
-        if (projectInspectorOutput.modules == null) {
-            return Collections.emptyList();
+        List<CodeLocation> codeLocations = new ArrayList<>();
+
+        if (outputFile == null || !outputFile.exists() || !outputFile.isFile()) {
+            logger.info("inspection.json file doesn't exist.");
+            return codeLocations;
         }
 
-        return projectInspectorOutput.modules.values().stream()
-            .map(this::codeLocationFromModule)
-            .collect(Collectors.toList());
+        //Utilized streaming parser to process the JSON file incrementally, without loading the entire file into memory.
+        //Memory-efficient for processing a large(400-500MB) JSON file.
+        try (JsonReader reader = new JsonReader(new FileReader(outputFile))) {
+            reader.beginObject();
+            while (reader.hasNext()) {
+                String moduleName = reader.nextName();
+                if (moduleName != null && moduleName.equals("Modules")) {
+                    reader.beginObject();
+                    while (reader.hasNext()) {
+                        String moduleId = reader.nextName();
+                        if (moduleId != null) {
+                            JsonObject module = new JsonParser().parse(reader).getAsJsonObject();
+                            ProjectInspectorModule projectInspectorModule = gson.fromJson(module, ProjectInspectorModule.class);
+                            codeLocations.add(codeLocationFromModule(projectInspectorModule));
+                        }
+                    }
+                    reader.endObject();
+                } else {
+                    reader.skipValue();
+                }
+            }
+            reader.endObject();
+        } catch (IOException e) {
+            logger.error("An error occurred while reading inspection.json file: {}", e.getMessage());
```

Comment:
e.getMessage() may not return the stacktrace. I prefer `logger.error("message", e);`